### PR TITLE
Implement add users directly inline from empty group view

### LINKED ISSUE #4883: Groups management UI
edit: (@viddo) split up this epic issue to smaller ones, to be able to implement this in incremental steps. Also ordered them in terms of necessarily (must-have and nice-to-have buckets) to prioritise and implement them in natural order of usage.

This issue will serve as a "parent" issue to keep track on overall progress (will create issues as starting to work/discuss details per use-case):

**"Must-haves"**
- Organisation owner can:
  - [x] create a new group #5277
  - [x] see groups belonging to organization #5277
  - [x] edit group name #5277
  - [x] delete group #5277
  - [x] group details header navigation (users | settings) #5311
  - [x] add users to a group #5431
  - [x] browse add-org-users list through pagination #5431
  - [x] search add-org-users #5431
  - [x] apply visual styles #5431
  - [x] remove users from a group #5431
- Any user can:
  - [x] [Share dataset with a group #4851](https://github.com/CartoDB/cartodb/issues/4851)

**"Nice-to-have"**
- Organisation owner can:
  - [x] see datasets/maps count for each group #5629
  - [x] see count/avatar previews of group members #5629
  - [x] see datasets/maps count for each group user #5629
  - [x] browse/search groups #5781 
  - [x] browse/search groups users #5913 
  - [x] add users to a new group (w/o members) directly from group view #5923
  - [x] Show empty-message when there are no groups #5650 

---

Relevant references:
- [(Invision) UI proposal](https://projects.invisionapp.com/share/P43R5D79K#/screens/93104597)

cc @saleiva

### LINKED ISSUE #5650: Show empty-message when there are no groups
screen shot 2015-09-22 at 11 35 17

cc @viddo

### LINKED ISSUE #5654: Add users to a new group (w/o members) directly from group view
screen shot 2015-09-22 at 11 57 43

cc @viddo

### PR DESCRIPTION
**Pull request by viddo**
resolves #5654 
resolves #5650 (by 5751f13)
…which thus finally closes #4883 :tada:

empty-group-add

Solved by having a parent view “on top” which manages the edge-cases of transitioning from/to empty group (i.e. no group users).

Also had to modify the group users collection to do a fetch on the batch operations to reflect the state (thus the promise resolve is delayed slightly).

@javierarce can you review please?
cc @xavijam @juanignaciosl

### CODE REVIEW

**Comment by xavijam** [comment]
Staging?

**Comment by viddo** [comment]
@xavijam :rocket: rui1/2 (hold your :horse: for a few minutes)

File: lib/assets/javascripts/cartodb/models/group_users.js [hunk 1]
```diff
@@ -39,13 +39,37 @@ cdb.admin.GroupUsers = Backbone.Collection.extend({
...
       data: {
         users: ids
+      },
+      success: function() {
+        var args = arguments;
+
+        // because add/remove don't return any data, so need to fetch to get accurate state
+        self.fetch({
+          success: function() {
+            deferred.resolve.apply(deferred, args)
```
Comment (javierarce) [inline lib/assets/javascripts/cartodb/models/group_users.js]: missing ;

File: lib/assets/javascripts/cartodb/models/group_users.js [hunk 2]
```diff
@@ -39,13 +39,37 @@ cdb.admin.GroupUsers = Backbone.Collection.extend({
...
+          },
+          error: function() {
+            // could not update state, but resolve anyway since batch operation worked
+            // might have inconsistent state though
+            deferred.resolve.apply(deferred, args)
```
Comment (javierarce) [inline lib/assets/javascripts/cartodb/models/group_users.js]: missing ;

**Comment by javierarce** [comment]
:+1: Code looks good!

**Comment by viddo** [comment]
@xavijam …deployed on rui1+2 now, have fun! :roller_coaster:

**Comment by xavijam** [comment]
Looks great!

**Pull request merged**